setup.py

```diff
@@ -1,48 +1,84 @@
-]
+# Check Python Version -----------------------------------------------------------------
+if sys.version_info < PY_VERSION:
+    print(
+        f"ERROR: {NAME} requires at least Python {PY_VERSION[0]}.{PY_VERSION[1]}+ to run."
+    )
+    sys.exit(1)
+
+
+# Helper Utilities ---------------------------------------------------------------------
+
+
+def _get_version() -> str:
+    with open(os.path.join(HERE, NAME, "__init__.py")) as init_f:
+        for line in init_f.readlines():
+            if "__version__" in line:
+                # grab "X.Y.Z" from "__version__ = 'X.Y.Z'" (quote-style insensitive)
+                return line.replace('"', "'").split("=")[-1].split("'")[1]
+    raise Exception("cannot find __version__")
+
+
+def _get_pypi_requirements() -> List[str]:
+    return [m.replace("==", ">=") for m in REQUIREMENTS if "git+" not in m]
+
+
+def _get_git_requirements() -> List[str]:
+    def valid(req: str) -> bool:
+        pat = r"^git\+https://github\.com/\w+/\w+@(v)?\d+\.\d+\.\d+#egg=\w+$"
+        if not re.match(pat, req):
+            raise Exception(
+                f"from {REQUIREMENTS_PATH}: "
+                f"pip-install git-package url is not in standardized format {pat} ({req})"
+            )
+        return True
+
+    return [m.replace("git+", "") for m in REQUIREMENTS if "git+" in m and valid(m)]
+
+
+# Setup --------------------------------------------------------------------------------
 
 setup(
-    name='file_catalog',
-    version=file_catalog.__version__,
-    description='File catalog',
-    long_description=long_description,
-    url='https://github.com/WIPACrepo/file_catalog',
-    license='MIT',
+    name=NAME,
+    version=_get_version(),
+    description="File Catalog",
+    long_description=open(os.path.join(HERE, "README.md")).read(),  # include new-lines
+    long_description_content_type="text/markdown",
+    url="https://github.com/WIPACrepo/file_catalog",
+    license="MIT",
     classifiers=[
-        'Development Status :: 4 - Beta',
-        'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
+        "Development Status :: 4 - Beta",
+        "License :: OSI Approved :: MIT License",
+        "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.3",
+        "Programming Language :: Python :: 3.4",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
```

Comment:
Should probably adjust these to match PY_VERSION

Reply:
do we need all the python version or just the current one?

Reply:
I'm including everything from 3.3+. I don't know why we are excluding 3.1-3.2, but I didn't touch that.

Reply:
I would only include the ones we expect it to work on.  So some of the early 3.x probably don't work anymore.

Reply:
Specifically, should python2 be removed now, especially if you start putting in f-strings?

Reply:
is there a way to test this? Otherwise, I'd just say 3.5

Reply:
Not easily?  You could probably make a bot try out installation/tests on each python version.  But that sounds painful.  You could also do "anything >= PY_VERSION" as a safe bet on what will work.

Reply:
I added a `OLDEST_PY_VERSION` tuple

Reply:
Yeah, Python 3.6 is minimum for f-strings